src/packages/frontend/course/students/students-panel.tsx

```diff
@@ -313,7 +315,7 @@ export const StudentsPanel: React.FC<StudentsPanelReactProps> = React.memo(
     }
 
     function student_add_button() {
-      const disabled = add_search?.trim().length === 0;
+      const disabled = add_search?.trim().length === 0 || !isValidEmailAddress(add_search?.trim() ?? "");
```

Comment:
I think the input is one or more email address, separated by whitespaces and/or commas, whereas above you are I think searching for only *ONE* email address.    Often an instructor pastes in a bunch of email addresses that they are copying out of a spreadsheet or somewhere else.